src/test/scala/is/hail/expr/ir/CompileSuite.scala

```diff
@@ -528,4 +526,52 @@ class CompileSuite {
     assert(expected.length == actual.length)
     assert(actual.sameElements(expected))
   }
+
+  def testArrayFlatMap() {
+    val tRange = TArray(TInt32())
+    val ir = ArrayFlatMap(ArrayRange(I32(0), In(0, TInt32()), I32(1)), "i", ArrayRange(I32(0), Ref("i"), I32(1)))
+    val region = Region()
+    val fb = FunctionBuilder.functionBuilder[Region, Int, Boolean, Long]
+    doit(ir, fb)
+    val f = fb.result()()
+    for {
+      stop <- 0 to 5
+    } {
+      val aoff = f(region, stop, false)
+      val expected = Array.range(0, stop, 1).flatMap(i => Array.range(0, i, 1))
+      val actual = new UnsafeIndexedSeq(tRange, region, aoff)
+      assert(expected.length == actual.length)
+      assert(actual.sameElements(expected))
```

Comment:
I see this was done all over CompileSuite, but `sameElements` already fails if they have different lengths.